src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/caseworker/event/page/SendOrReply.java

```diff
@@ -0,0 +1,146 @@
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.ccd.sdk.type.DynamicList;
+import uk.gov.hmcts.ccd.sdk.type.DynamicListElement;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.CaseData;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.State;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.SelectedMessage;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.MessageDocumentList;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.MessageSendDetails;
+import uk.gov.hmcts.reform.adoption.common.ccd.CcdPageConfiguration;
+import uk.gov.hmcts.reform.adoption.common.ccd.PageBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+public class SendOrReply implements CcdPageConfiguration {
+    @Override
+    public void addTo(PageBuilder pageBuilder) {
+        pageBuilder
+            .page("pageSendOrReply1", this::midEvent)
+            .mandatory(CaseData::getMessageAction)
+            .mandatory(CaseData::getReplyMsgDynamicList, "messageAction=\"replyMessage\"")
+            .page("pageSendOrReply3")
+            .showCondition("messageAction=\"replyMessage\"")
+            .label("labelReplyMes", "## Reply a message")
+            .complex(CaseData::getSelectedMessage)
+            .readonly(SelectedMessage::getReasonForMessage)
+            .readonly(SelectedMessage::getUrgency)
+            .readonly(SelectedMessage::getMessage)
+            .readonly(SelectedMessage::getDocumentLink)
+            .mandatory(SelectedMessage::getReplyMessage)
+            .done()
+            .page("pageSendOrReply2")
+            .showCondition("messageAction=\"sendMessage\"")
+            .label("sendMessage","## Send a message")
+            .complex(CaseData::getMessageSendDetails)
+            .mandatory(MessageSendDetails::getMessageReceiverRoles)
+            .mandatory(MessageSendDetails::getMessageReasonList)
+            .mandatory(MessageSendDetails::getMessageUrgencyList)
+            .mandatory(MessageSendDetails::getMessage)
+            .done()
+            .mandatory(CaseData::getSendMessageAttachDocument)
+            .mandatory(CaseData::getAttachDocumentList, "sendMessageAttachDocument=\"Yes\"")
+            .done();
+    }
+
+
+    private AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseData, State>
+        data, CaseDetails<CaseData, State> caseDataStateCaseDetails1) {
+        log.info("MidEvent Triggered");
+        CaseData caseData = data.getData();
+        List<DynamicListElement> listElements = new ArrayList<>();
+        List<MessageDocumentList> messageDocumentLists = new ArrayList<>();
+        if (caseData.getAdditionalDocumentsCategory() != null && caseData.getApplicationDocumentsCategory().size() > 0) {
```

Comment:
Can we use 'CollectionUtils.isNotEmpty()' instead of checking 2 diff conditions?

Reply:
Done